Optimize group_norm on cpu

xiaomengy opened the pull request:
Summary: Optimize group_norm on cpu

Differential Revision: D9177878

houseroad reviewed (approved):
LGTM, nit in the inline comments

BIT-silence commented on caffe2/operators/group_norm_op.cc:
```diff
@@ -18,29 +18,64 @@ namespace caffe2 {
+  const int C = G * D;
+  EigenArrayMap<T>(Y, D * HxW, N * G) =
+      (ConstEigenArrayMap<T>(X, D * HxW, N * G).rowwise() -
+       ConstEigenVectorArrayMap<T>(mu, N * G).transpose())
```
I have tested them, there is almost no difference for transposing a col vector.

xiaomengy committed:
Optimize group_norm on cpu (#10246)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/10246

Optimize group_norm on cpu

Reviewed By: houseroad

Differential Revision: D9177878

fbshipit-source-id: 581ac9d90deff385813b50b19e523c254718506c
